source/drivers/AnalogSensor.cpp

```diff
@@ -46,196 +46,17 @@ using namespace codal;
-
-    return DEVICE_OK;
-}
-
-/**
- * Reads the currently configured sample period.
- *
- * @return The time between samples, in milliseconds.
- */
-int AnalogSensor::getPeriod()
-{
-    return samplePeriod;
-}
-
-/**
- * Set threshold to the given value. Events will be generated when these thresholds are crossed.
- *
- * @param value the LOW threshold at which a ANALOG_THRESHOLD_LOW will be generated.
- *
- * @return DEVICE_OK on success, DEVICE_INVALID_PARAMETER if the request fails.
- */
-int AnalogSensor::setLowThreshold(uint16_t value)
-{
-    // Protect against churn if the same threshold is set repeatedly.
-    if ((status & ANALOG_SENSOR_LOW_THRESHOLD_ENABLED) && lowThreshold == value)
-        return DEVICE_OK;
-
-    // We need to update our threshold
-    lowThreshold = value;
-
-    // Reset any exisiting threshold state, and enable threshold detection.
-    status &= ~ANALOG_SENSOR_LOW_THRESHOLD_PASSED;
-    status |=  ANALOG_SENSOR_LOW_THRESHOLD_ENABLED;
-
-    // If a HIGH threshold has been set, ensure it's above the LOW threshold.
-    if(status & ANALOG_SENSOR_HIGH_THRESHOLD_ENABLED)
-        setHighThreshold(max(lowThreshold+1, highThreshold));
-
-    return DEVICE_OK;
-}
-
-/**
- * Set threshold to the given value. Events will be generated when these thresholds are crossed.
- *
- * @param value the HIGH threshold at which a ANALOG_THRESHOLD_HIGH will be generated.
- *
- * @return DEVICE_OK on success, DEVICE_INVALID_PARAMETER if the request fails.
- */
-int AnalogSensor::setHighThreshold(uint16_t value)
-{
-    // Protect against churn if the same threshold is set repeatedly.
-    if ((status & ANALOG_SENSOR_HIGH_THRESHOLD_ENABLED) && highThreshold == value)
-        return DEVICE_OK;
-
-    // We need to update our threshold
-    highThreshold = value;
-
-    // Reset any exisiting threshold state, and enable threshold detection.
-    status &= ~ANALOG_SENSOR_HIGH_THRESHOLD_PASSED;
-    status |=  ANALOG_SENSOR_HIGH_THRESHOLD_ENABLED;
-
-    // If a HIGH threshold has been set, ensure it's above the LOW threshold.
-    if(status & ANALOG_SENSOR_LOW_THRESHOLD_ENABLED)
-        setLowThreshold(min(highThreshold - 1, lowThreshold));
-
-    return DEVICE_OK;
-}
-
-/**
- * Determines the currently defined low threshold.
- *
- * @return The current low threshold. DEVICE_INVALID_PARAMETER if no threshold has been defined.
- */
-int AnalogSensor::getLowThreshold()
-{
-    if (!(status & ANALOG_SENSOR_LOW_THRESHOLD_ENABLED))
-        return DEVICE_INVALID_PARAMETER;
-
-    return lowThreshold;
-}
-
-/**
- * Determines the currently defined high threshold.
- *
- * @return The current high threshold. DEVICE_INVALID_PARAMETER if no threshold has been defined.
- */
-int AnalogSensor::getHighThreshold()
-{
-    if (!(status & ANALOG_SENSOR_HIGH_THRESHOLD_ENABLED))
-        return DEVICE_INVALID_PARAMETER;
-
-    return highThreshold;
+    return _pin.getAnalogValue();
```

Comment:
Cool - so a new driver would basically just be a constructor and this function? NICE!

Reply:
Yes you can see in classes `LinearAnalogSensor` and `NonLinearAnalogSensor` that just use readvalue thus the encapsulation is preserved !